caf-audit/src/test/java/com/hpe/caf/auditing/elastic/ElasticAuditIT.java

```diff
@@ -0,0 +1,337 @@
+    }
+
+    @Test(expected = Exception.class)
+    public void testUnknownESHost() throws Exception {
+
+        final String esUnknownHostName = "unknown";
+
+        try (
+                AuditConnection auditConnection = AuditConnectionFactory.createConnection(new ConfigurationSource() {
+
+                    @Override
+                    public <T> T getConfiguration(Class<T> aClass) throws ConfigurationException {
+                        List<String> hostNames = new ArrayList<>();
+                        hostNames.add(esUnknownHostName);
+
+                        ElasticAuditConfiguration elasticAuditConfiguration = new ElasticAuditConfiguration();
+                        elasticAuditConfiguration.setHostnames(hostNames);
+                        elasticAuditConfiguration.setPort(ES_PORT);
+                        elasticAuditConfiguration.setClusterName(ES_CLUSTERNAME);
+                        return (T) elasticAuditConfiguration;
+                    }
+                })
+        ) {
+            // Do nothing as exception is expected to be thrown.
+        }
+    }
+
+    @Test(expected = Exception.class)
+    public void testIncorrectESPort() throws Exception {
+
+        final int esUnexpectedPort = 9100;
+
+        try (
+                AuditConnection auditConnection = AuditConnectionFactory.createConnection(new ConfigurationSource() {
+
+                    @Override
+                    public <T> T getConfiguration(Class<T> aClass) throws ConfigurationException {
+                        List<String> hostNames = new ArrayList<>();
+                        hostNames.add(ES_HOSTNAME);
+
+                        ElasticAuditConfiguration elasticAuditConfiguration = new ElasticAuditConfiguration();
+                        elasticAuditConfiguration.setHostnames(hostNames);
+                        elasticAuditConfiguration.setPort(esUnexpectedPort);
+                        elasticAuditConfiguration.setClusterName(ES_CLUSTERNAME);
+                        return (T) elasticAuditConfiguration;
+                    }
+                });
+                com.hpe.caf.auditing.AuditChannel auditChannel = auditConnection.createChannel()
+        ) {
+            //  Index a sample audit event message into Elasticsearch.
+            AuditEventBuilder auditEventBuilder = auditChannel.createEventBuilder();
+
+            //  Set up fixed field data for the sample audit event message.
+            auditEventBuilder.setApplication(APPLICATION_ID);
+            auditEventBuilder.setEventType(EVENT_CATEGORY_ID, EVENT_TYPE_ID);
+            auditEventBuilder.setCorrelationId(CORRELATION_ID);
+            auditEventBuilder.setTenant(TENANT_ID);
+            auditEventBuilder.setUser(USER_ID);
+
+            //  No need to set up custom data as we expect the call to index the document to fail because
+            //  of the unexpected port used.
+            auditEventBuilder.send();
```

Comment:
In this test, we do not check for the audit event in Elasticsearch because the .prepareIndex call is expected to fail with port 9100 usage. This will throw an exception in the code at this point. The expected exception is flagged in the @Test annotation.